src/i18n/languageApi.js

```diff
@@ -0,0 +1,54 @@
+import { getConfig } from '../config';
+import { getAuthenticatedHttpClient } from '../auth';
+import { convertKeyNames, snakeCaseObject } from '../utils';
+
+/**
+ * Updates user language preferences via the preferences API.
+ *
+ * This function converts preference data to snake_case and formats specific keys
+ * according to backend requirements before sending the PATCH request.
+ *
+ * @param {string} username - The username of the user whose preferences to update.
+ * @param {Object} preferenceData - The preference parameters to update (e.g., { prefLang: 'en' }).
+ * @returns {Promise} - A promise that resolves when the API call completes successfully,
+ *                      or rejects if there's an error with the request.
+ */
+export async function updateUserPreferences(username, preferenceData) {
```

Comment:
As suggested in another review comment, I would change this function like so:

```suggestion
export async function updateAuthenticatedUserPreferences(preferenceData) {
```

And then get the authenticated username inside the function.  Unless, of course, you can imagine a situation where we'd need this function to change the preference of some user other than the currently logged in one - but I find that to be unlikely.